eventing/samples/gcp-pubsub-source/README.md

```diff
@@ -52,17 +52,23 @@ source is most useful as a bridge from other GCP services, such as
       gcloud iam service-accounts keys create knative-source.json \
         --iam-account=knative-source@$PROJECT_ID.iam.gserviceaccount.com
       ```
-   1. Create a secret on the kubernetes cluster for the downloaded key. You need
-      to store this key in `key.json` in a secret named `gcppubsub-source-key`
-
+   1. Create two secrets on the kubernetes cluster with the downloaded key:
       ```shell
-      kubectl -n knative-sources create secret generic gcppubsub-source-key --from-file=key.json=knative-source.json
+      # Note that the first secret may already have been created when installing
+      # Knative Eventing. The following command will overwrite it. If you don't
+      # want to overwrite it, then skip this command.
+      kubectl -n knative-sources create secret generic gcppubsub-source-key --from-file=key.json=knative-source.json --dry-run -o yaml | kubectl apply --filename -
```

Comment:
I presume you're doing this this way because the secret may previously have been managed by `kubectl apply`?

Reply:
> I presume you're doing this this way because the secret may previously have been managed by `kubectl apply`?

No. I do this because it easily handles the create or replace with a single command.

I spent a fair amount of time trying to figure out how to create the secret if it wasn't there, or replace the secret if it was. `kubectl create secret` has the really useful --from-file flag, but `kubectl replace` doesn't, which immediately made the replace case much harder. My initial plan was to use a shell script along the lines of:

```shell
if secret exists; then
  delete secret
fi
create secret
```

I also thought about getting the secret, using `jq` to replace the secret value and then apply it again, but didn't want to add a dependency on `jq`.

Then I ran across this dry-run->apply pattern and saw that it nicely handled my problem. Namely, it will either create or replace the secret to the exact value I want. It does produce a warning if the original was not created in the proper way, but modifies it correctly:

```
Warning: kubectl apply should be used on resource created by either kubectl create --save-config or kubectl apply
```